presentation/testing/src/test/IModelUtilities.test.ts

```diff
@@ -0,0 +1,174 @@
+  let codeSpecMock: moq.IMock<CodeSpec>;
+
+  let modelPropsMock: moq.IMock<ModelProps>;
+  let elementPropsMock: moq.IMock<ElementProps>;
+  let elementAspectPropsMock: moq.IMock<ElementAspectProps>;
+  let codeScopePropsMock: moq.IMock<CodeScopeProps>;
+
+  let snapshotDbMock: moq.IMock<SnapshotDb>;
+  let snapshotConnectionMock: moq.IMock<SnapshotConnection>;
+
+  afterEach(() => {
+    sinon.restore();
+  });
+
+  const setupSnapshot = () => {
+    snapshotDbMock = moq.Mock.ofType<SnapshotDb>();
+    sinon.stub(SnapshotDb, "createEmpty").returns(snapshotDbMock.object);
+    sinon.stub(SnapshotConnection, "openFile");
+  };
+
+  it("insertModel calls iModel.models.insertModel", async () => {
+    imodelMock = moq.Mock.ofType<IModelDb>();
+    modelPropsMock = moq.Mock.ofType<ModelProps>();
+    modelsMock = moq.Mock.ofType<IModelDb.Models>();
+    imodelMock.setup((x) => x.models).returns(() => modelsMock.object);
+
+    const builder = new IModelBuilder(imodelMock.object);
+    builder.insertModel(modelPropsMock.object);
+
+    modelsMock.verify(async (x) => x.insertModel(modelPropsMock.object), moq.Times.once());
+  });
+
+  it("insertElement calls iModel.models.insertElement", async () => {
+    imodelMock = moq.Mock.ofType<IModelDb>();
+    elementPropsMock = moq.Mock.ofType<ElementProps>();
+    elementsMock = moq.Mock.ofType<IModelDb.Elements>();
+    imodelMock.setup((x) => x.elements).returns(() => elementsMock.object);
+
+    const builder = new IModelBuilder(imodelMock.object);
+    builder.insertElement(elementPropsMock.object);
+
+    elementsMock.verify(async (x) => x.insertElement(elementPropsMock.object), moq.Times.once());
+  });
+
+  it("insertAspect calls iModel.models.insertAspect", async () => {
+    imodelMock = moq.Mock.ofType<IModelDb>();
+    elementAspectPropsMock = moq.Mock.ofType<ElementAspectProps>();
+    elementsMock = moq.Mock.ofType<IModelDb.Elements>();
+    imodelMock.setup((x) => x.elements).returns(() => elementsMock.object);
+
+    const builder = new IModelBuilder(imodelMock.object);
+    builder.insertAspect(elementAspectPropsMock.object);
+
+    elementsMock.verify(async (x) => x.insertAspect(elementAspectPropsMock.object), moq.Times.once());
+  });
+
+  it("createCode calls iModel.codeSpecs.getByName", () => {
+    imodelMock = moq.Mock.ofType<IModelDb>();
+    codeSpecsMock = moq.Mock.ofType<CodeSpecs>();
+    codeScopePropsMock = moq.Mock.ofType<CodeScopeProps>();
+    codeSpecMock = moq.Mock.ofType<CodeSpec>();
+    codeSpecsMock.setup((x) => x.getByName(BisCodeSpec.drawing)).returns(() => codeSpecMock.object).verifiable(moq.Times.once());
+    imodelMock.setup((x) => x.codeSpecs).returns(() => codeSpecsMock.object);
+
+    const builder = new IModelBuilder(imodelMock.object);
+    builder.createCode(codeScopePropsMock.object, BisCodeSpec.drawing, "codeValue");
+
+    codeSpecsMock.verifyAll();
+  });
+
+  it("calls IModelJsFs.mkdirSync if directory does not exist", async () => {
```

Comment:
Please group the tests based on what they're testing. E.g.
```ts
describe("IModelBuilder", () => {

  // IModelBuilder tests

});

describe("buildTestIModel", () => {

  // buildTestIModel tests

});
```

Reply:
Done